src/nlsolve/utils.jl

```diff
@@ -20,8 +20,14 @@ function assess_convergence(x,
         x_converged = true
     end
 
-    if maximum(abs, f) <= ftol
-        f_converged = true
+    if ftol isa AbstractArray
+        if all( f .< ftol )
```

Comment:
Missing an `abs` there, and should be `.<=` for consistency. I think it's fine to keep this even for scalar `ftol`, even though it allocates more than `maximum(abs,f)<= tol`.

Reply:
Ok, I wasn't sure what the preferences are. I changed it accordingly